### ISSUE TITLE
Tests of v3.1 fail on aarch64

### ISSUE THREAD

**Issue by ignatenkobrain** [Nitrokey/libnitrokey#76]
```
1: Test command: /builddir/build/BUILD/libnitrokey-3.1/aarch64-redhat-linux-gnu/test_offline
1: Test timeout computed to be: 1500
1: [Tue Oct 17 09:44:44 2017][DEBUG_L1]	Throw: Device not initialized
1: [Tue Oct 17 09:44:44 2017][DEBUG_L1]	Throw: Device not initialized
1: [Tue Oct 17 09:44:44 2017][DEBUG_L1]	Throw: Device not initialized
1: 
1: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1: test_offline is a Catch v1.10.0 host application.
1: Run with -? for options
1: 
1: -------------------------------------------------------------------------------
1: Test helper function - hex_string_to_byte
1: -------------------------------------------------------------------------------
1: /builddir/build/BUILD/libnitrokey-3.1/unittest/test_offline.cc:89
1: ...............................................................................
1: 
1: /builddir/build/BUILD/libnitrokey-3.1/unittest/test_offline.cc:97: FAILED:
1:   REQUIRE( v.data()[i] == test_data[i] )
1: with expansion:
1:   11 == 0
1: with message:
1:   Position i: 0
1: 
1: ===============================================================================
1: test cases:  4 |  3 passed | 1 failed
1: assertions: 53 | 52 passed | 1 failed
1: 
1/1 Test #1: runs .............................***Failed    0.01 sec
```

I think I've seen this before and it works after few tries so might be some race condition or so.

```cxx
[ 91%] Building CXX object CMakeFiles/test_offline.dir/unittest/test_offline.cc.o
/usr/bin/c++   -I/builddir/build/BUILD/libnitrokey-3.1/hidapi -I/builddir/build/BUILD/libnitrokey-3.1/include -I/builddir/build/BUILD/libnitrokey-3.1/unittest/Catch/include  -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -O2 -g -DNDEBUG   -std=gnu++14 -o CMakeFiles/test_offline.dir/unittest/test_offline.cc.o -c /builddir/build/BUILD/libnitrokey-3.1/unittest/test_offline.cc
make[2]: Leaving directory '/builddir/build/BUILD/libnitrokey-3.1/aarch64-redhat-linux-gnu'
/builddir/build/BUILD/libnitrokey-3.1/unittest/test_offline.cc: In function 'void ____C_A_T_C_H____T_E_S_T____5()':
/builddir/build/BUILD/libnitrokey-3.1/unittest/test_offline.cc:95:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i = 0; i < v.size(); ++i) {
                   ~~^~~~~~~~~~
```

^ might be related as well

**Comment by szszszsz** [comment]
Is the fail message always the same? That is, is position `i` changing in this test?
My guess is this is related to big endian/little endian. But it is not summing up with passing it sometimes (should always fail in this case).

As for compilation warning - this one should not be a cause this time, but I will fix it to not clutter the log.

**Comment by ignatenkobrain** [comment]
@szszszsz aarch64 seems to be supporting both BE and LE, but I think it is executed as LE although not sure. I resubmitted build and it passed now. Last time it failed it was also `0`.

**Comment by szszszsz** [comment]
Fixed in #96
